yt_dlp/extractor/sharepoint.py

```diff
@@ -0,0 +1,65 @@
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    float_or_none,
+    traverse_obj,
+)
+
+
+class SharePointIE(InfoExtractor):
+    _VALID_URL = r'https?://[^\.]+\.sharepoint\.com/:v:/g/[^/]+/[^/]+/(?P<id>[^/?]+)'
+
+    _TESTS = [
+        {
+            'url': 'https://lut-my.sharepoint.com/:v:/g/personal/juha_eerola_student_lab_fi/EUrAmrktb4ZMhUcY9J2PqMEBD_9x_l0DyYWVgAvp-TTOMw?e=ZpQOOw',
+            'info_dict': {
+                'id': 'EUrAmrktb4ZMhUcY9J2PqMEBD_9x_l0DyYWVgAvp-TTOMw',
+                'ext': 'mp4',
+                'title': 'CmvpJST.mp4',
+                'duration': 54.567,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+    ]
+
+    def _parse_mediapackage(self, video, video_id):
+        formats = []
+
+        base_media_url = video['.transformUrl'].replace('/thumbnail?', '/videomanifest?')
+        base_media_url += f"&cTag={video['.ctag']}&action=Access&part=index"
+
+        metadata = json.loads(video.get('MediaServiceFastMetadata'))
```

Comment:
* This is still fatal and gets no benefit from the `.get`. Use `self._parse_json` instead
* Move it below near where it's actually used